ivre/db/__init__.py

```diff
@@ -2467,16 +2465,135 @@ def _get_timeslots(cls, start_time, end_time):
         Returns an array of timeslots included between start_time and end_time
         """
         times = []
-        time = cls.date_round(start_time)
-        end_timeslot = cls.date_round(end_time)
-        while time <= end_timeslot:
+        first_timeslot = cls._get_timeslot(
+            start_time, config.FLOW_TIME_PRECISION, config.FLOW_TIME_BASE
+        )
+        time = first_timeslot['start']
+        last_timeslot = cls._get_timeslot(
+            end_time, config.FLOW_TIME_PRECISION, config.FLOW_TIME_BASE
+        )
+        end_time = last_timeslot['start']
+        while time <= end_time:
             d = OrderedDict()
```

Comment:
In Python, each timeslot is a dictionary like `{'start': <datetime>, 'duration': <int>}`. Dictionary type does not exist in MongoDB. Thus, before being inserted in MongoDB, these dictionaries are transformed in `Object` by `pymongo`. 

Timeslots are stored as an `Array` in MongoDB. To insert/update timeslots, I use the MongoDB operator `$addToSet` which ensures the unicity of elements. `Object` elements are sensible to the attributes order, meaning that `{a: 1, b: 2}` is different than `{b: 2, a: 1}`. 

Because in Python the order of fields in a dictionary is undefined, I must use an OrderedDict so that I don't get duplicated timeslots.

PS : It is to be noted that Python 2.6 does not support `OrderedDict`. The fallback is to use a classic `dict`. According to Travis builds (actually the absence of random fails for Python 2.6), it seems that Python 2.6 dicts are ordered by default.

Reply:
Perfect. Could you add this explanation as a comment in the code?